### ISSUE TITLE
Can't reproduce performance of the SRL model used in the demo on CoNLL 2012 dataset

### ISSUE THREAD

**Issue by chaojiang06** [allenai/allennlp#4220]
**System (please complete the following information):**
 - OS: Ubuntu 18.04.3 LTS
 - Python version: 3.7
 - AllenNLP version: v1.0.0.rc3
 - PyTorch version: 1.5

**Question**
Hi, I try to reproduce the results on the OntoNotes dataset (conll 2012) in the [Shi et al., 2019](https://arxiv.org/pdf/1904.05255.pdf) paper used in the SRL [demo](https://demo.allennlp.org/semantic-role-labeling). However, I could only get F1 around 0.79.

***Command I used***:
`allennlp evaluate https://storage.googleapis.com/allennlp-public-models/bert-base-srl-2020.03.24.tar.gz /data/conll-formatted-ontonotes-5.0/conll-formatted-ontonotes-5.0-12/conll-formatted-ontonotes-5.0/data/conll-2012-test/data/english`

***Pre-processing the dataset***
I tried the pre-processing methods mentioned in both [here](http://conll.cemantix.org/2012/data.html) and [here](http://cemantix.org/data/ontonotes.html), I also tried both `conll-2012-test` (25624 sentences) and `test` (29669 sentences) folders to evaluate. In all cases, I can only get F1 around 0.794.

I was wondering should the F1 be 86.2 for `BERT-LSTM-base` as described in Table 5 [here](https://arxiv.org/pdf/1904.05255.pdf#page=4)? 

I can think of three potential reasons:
1. Pre-processing. 
2. How the predicted is specified.  The paper reports the performance when gold predicates are given. Is this the same as the current evaluation method? Matt's [answer](https://github.com/allenai/allennlp/issues/4160#issuecomment-622596480) says the predicates are found by the POS tagger in Spacy package. 
3. The pre-trained SRL model is implemented by code in [srl_bert_model](https://github.com/allenai/allennlp-models/blob/master/allennlp_models/syntax/srl/srl_bert_model.py), which seems to be different from what described in the [paper](https://arxiv.org/pdf/1904.05255.pdf#page=2), as there is no BiLSTM after the BERT embedding.  

***A second question***
The code in [srl_bert_model](https://github.com/allenai/allennlp-models/blob/master/allennlp_models/syntax/srl/srl_bert_model.py) seems to be different from what described in the [paper](https://arxiv.org/pdf/1904.05255.pdf#page=3), as there is no BiLSTM after the BERT embedding. I was wondering do you have implementation for the `BERT w/ BiLSTM` model? Or the current implementation can achieve 86.2 F1?

Any help would be appreciated!

**Comment by matt-gardner** [comment]
When that model was developed, it achieved 86.49 F1, as it says on our demo page.  I would be *very* surprised if there was a code change that somehow made performance go down; more likely it is one of your first two guesses.  I know there is some preprocessing that goes into producing the training and evaluation data, but I don't know what it is.  Maybe @DeNeutoy could answer that?

Oh, another possibility - I know there's a separate evaluation script that you have to run to get the actual metrics.  I don't know if what you see during `allennlp evaluate` is actually the right metric or not.  That's another question that @DeNeutoy could answer better than me.

**Comment by chaojiang06** [comment]
Thank you for the response! 

Hi @DeNeutoy, I also tried to use the `jsonnet` from [here](https://github.com/allenai/allennlp-models/blob/master/training_config/syntax/bert_base_srl.jsonnet) to train the SRL model on ontonotes dataset from scratch. the best performance I can get is 0.795 F1 on dev set and 0.797 on test set. 

The command I used is `allennlp train   /data/bert_base_srl.jsonnet  -s /scratch/trial_4 --include-package allennlp_models`. I only change batch size to 16 to fit in my GPU memory

I compare my output with yours at [here](https://github.com/allenai/allennlp/issues/3395#issuecomment-546032387). We have the same number of sentences in training / development set (278026 / 38377 sentences). So I guess the data should be fine. However, your dev performance looks very good, starts at 0.795, and reaches 0.821 at the second epoch. In contrast, my dev performance start around 60 F1 and never go beyond 80 F1. 

I was wondering what causes a huge gap? The model structure or the evaluation? The model in the [code](https://github.com/allenai/allennlp-models/blob/master/allennlp_models/syntax/srl/srl_bert_model.py) looks really different from that described in the [paper](https://arxiv.org/pdf/1904.05255.pdf#page=3) to me, as there is no BiLSTM layer. Current evaluation method should already use `srl-eval.pl` at somewhere. 

Thank you in advance!

**Comment by DeNeutoy** [comment]
@chaojiang06 please can you try with the latest stable release of allennlp, e.g 0.9. It's possible that something from the huggingface optimizer code changed (e.g the "huggingface_adamw" name for the optimizer might have changed?).

Yes, it is even simpler than in that paper - it doesn't use the LSTMs on purpose. 


Please try again with allennlp v0.9 and also let me know any changes at all that you made to the jsonnet config.

**Comment by chaojiang06** [comment]
Hi, a quite update on the evaluation using AllenNLP 0.9, the model `https://s3-us-west-2.amazonaws.com/allennlp/models/bert-base-srl-2019.06.17.tar.gz` can achieve 0.865 F1, but the latest model `https://storage.googleapis.com/allennlp-public-models/bert-base-srl-2020.03.24.tar.gz` run into an error.

I am trying to train a model from scratch, it looks good until now.

**Comment by chaojiang06** [comment]
One more update about the training, using the `jsonnet` file extracted from the model `https://s3-us-west-2.amazonaws.com/allennlp/models/bert-base-srl-2019.06.17.tar.gz` to train the model, I can reproduce the 86.4 F1 on the conll-2012-test testset.

**Comment by edchengg** [comment]
I can't reproduce this result with allennlp=1.0.0 using the same command: 

`allennlp evaluate https://storage.googleapis.com/allennlp-public-models/bert-base-srl-2020.03.24.tar.gz /data/conll-formatted-ontonotes-5.0/conll-formatted-ontonotes-5.0-12/conll-formatted-ontonotes-5.0/data/conll-2012-test/data/english`

Not able to run evaluate it with allennlp==0.9 because the allennlp-models ==1.0
